docs/site/pages/plugins/async-node.mdx

```diff
@@ -0,0 +1,122 @@
+
+In the example below, node with the id "some-async-node" will not be rendered on first render, but will be replaced with a concrete asset node at a later time:
+
+```json
+{
+  "id": "flow-1",
+    "views": [
+      {
+        "id": 'action',
+        "actions": [
+          {
+            "id": "some-async-node",
+            "async": true,
+          },
+        ],
+      },
+    ],
+  ...
+}
+```
+
+The `AsyncNodePlugin` exposes an `onAsyncNode` hook on all platforms. The `onAsyncNode` hook will be invoked with the current node when the plugin is available and an `AsyncNode` is detected during the resolve process. The node used to call the hook with could contain metadata according to content spec.
+
+User should tap into the `onAsyncNode` hook to examine the node's metadata before making a decision on what to replace the async node with. The return could be a single node or an array of nodes.
+
+
+### Continuous Streaming
+
+In order to keep streaming in new content, there must be at least 1 or more `AsyncNode`s in the view tree at all times.    
+This means there must be a constant renewal of new `AsyncNode`s after the previous ones are resolved by the user.
+
+## Usage
+
+<PlatformTabs>
+  <core>
+
+Add the plugin to Player:
+
+```js
+import { Player } from '@player-ui/player';
+import { AsyncNodePlugin } from '@player-ui/async-node-plugin';
+
+const asyncNodePlugin = new AsyncNodePlugin();
+
+// Configuring async node behaviour
+asyncNodePlugin.hooks.onAsyncNode.tap('handleAsync', async (node: Node.Node) => {
+  ...
+  // Determine what to return to be parsed into a concrete node
+});
+
+const player = new Player({
+  plugins: [
+    asyncNodePlugin
+  ]
+})
+```
+  </core>
+  <react>
+
+The `react` version of the AsyncNodePlugin is identical to using the core plugin. Refer to core usage for handler configuration:
+
+```js
+import { ReactPlayer } from '@player-ui/react';
+import { MetricsPlugin } from '@player-ui/async-node-plugin';
+
+const asyncNodePlugin = new AsyncNodePlugin();
+
+const player = new ReactPlayer({
+  plugins: [
+    asyncNodePlugin
+  ]
+})
+```
+
+  </react>
+  <ios>
+
+  </ios>
+  <android>
+
+In build.gradle
+```kotlin
+implementation "com.intuit.player.plugins:async-node:$JVM_PLAYER_VERSION"
+```
+
+In integration code
+```kotlin
+import com.intuit.player.plugins.asyncnode.AsyncNodePlugin
+
+val asyncNodePlugin = AsyncNodePlugin()
+
+// Configuring async node behaviour
+asyncNodePlugin.hooks.onAsyncNode.tap("handleAsync") { hookContext, node ->
+    ...
+    // Determine what to return in the form of a list of maps representing nodes that can be parsed
+    // e.g.
+    // listOf(
+    //  mapOf(
+    //    "asset" to mapOf(
+    //      "id" to "asset-1",
+    //      "type" to "text",
+    //      "value" to "new asset!"
+    //    )
+    //  )
+    // )
```

Comment:
I wouldn't put this in terms of nodes -- that means little to consumers. Really, what they're interested in is UI, so asset representation. This goes for the above docs too. I'd just put it in terms of content, which is really what you have here, but it's worth calling out.

Maybe there can be a sentence or two calling out that this'd work (?) for replacing non-asset nodes (like metadata or something), if that's applicable. Not sure there is a great use case for that though.